app/controllers/users/sessions_controller.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+class Users::SessionsController < Devise::SessionsController
+  before_action :configure_sign_in_params, only: [:create]
+
+  SIGN_IN_ERRORS = Struct.new(:username_or_email, :password)
+
+  SIGN_ING_STRUCT = Struct.new(:username_or_email, :password, :errors) do
```

Comment:
It's odd to see a Struct definition being saved in a constant. You can just define a class called SignInErrors instead of SIGN_IN_ERRORS.

Also it looks like SIGN_ING_STRUCT isn't being used anywhere
```suggestion
  SignInErrors = Struct.new(:username_or_email, :password)
```

Reply:
Actually I just realized SIGN_IN_ERRORS is being used inside the definition of SIGN_ING_STRUCT, but SIGN_ING_STRUCT is not being used anywhere. Do you really need this code at all?

Reply:
uffff thank you, no no, nothing of that is implemented, just one thing I was playing with.

Reply:
removed